semgrep/semgrep/rule_match.py

```diff
@@ -63,6 +65,18 @@ def start(self) -> Dict[str, Any]:
     def end(self) -> Dict[str, Any]:
         return self._end
 
+    @property
+    def lines(self) -> Iterable[str]:
+        """
+            Return lines in file that this RuleMatch is referring to.
+
+            Assumes file exists.  Note that start/end line is one-indexed
+        """
+        with self.path.open(
```

Comment:
could possible add `@lru_cache()` if we expect this to be read multiple times. I noticed that black does this for reading .gitignore

Reply:
I don't expect this for now but good to know this exists